lib/Appinfo/Application.php

```diff
@@ -25,16 +25,32 @@
 namespace OCA\Files3d\AppInfo;
 
 use OCA\Viewer\Event\LoadViewer;
-use OCA\Viewer\Listener\LoadViewerScript;
+use OCA\Files3d\Listener\LoadFiles3dScript;
 use OCP\AppFramework\App;
 use OCP\EventDispatcher\IEventDispatcher;
+use OCP\Files\IMimeTypeDetector;
 
 class Application extends App {
-    public function __construct() {
-        parent::__construct('files_3d');
+
+    const APP_ID = 'files_3d';
+    private $mimeTypeDetector;
+
+    public function __construct(IMimeTypeDetector $mimeTypeDetector) {
+        parent::__construct(self::APP_ID);
+
+        $this->mimeTypeDetector = $mimeTypeDetector;
 
         $server = $this->getContainer()->getServer();
         $eventDispatcher = $server->query(IEventDispatcher::class);
```

Comment:
Can also be injected similarely as IMimeTypeDetector

Reply:
(I think I should cleanup the viewer code too) :see_no_evil:

Reply:
> SHould also be injected similarely as IMimeTypeDetector :)
> And then move the addServiceListener to the register function I guess :)

Still doesn't work :thinking: 
Should the script be loaded on loading any page of my nc instance? or only when opening a supported file

Reply:
> Should the script be loaded on loading any page of my nc instance? or only when opening a supported file



The script will be loaded on every app that request the Viewer.
So it should be loaded on files

Reply:
I'm looking

Reply:
:grimacing: I just realized the app has been disabled :sweat_smile: 
But when enabling, it's disabled again (without an error). In the logs I get a `Could not resolve OCA\\Files3d\\AppInfo\\Application! Class OCA\\Files3d\\AppInfo\\Application does not exist` in my app.php. But at least for me it exists. phpstorm is on my side :D
Do you have any ideas?

Reply:
> I just realized the app has been disabled

Yes, I fixed it, wait for my push :)
If an app fails hard, it gets disabled as a prevention :)

Reply:
Ah, I don't have permissions, can you invite me as a contributor? :)

Reply:
makes sense.

But any idea why it fails in the first place? :D

Reply:
Yes, I fixed it, You'll see in the commits once I can push to your branch :)

Reply:
Perfect! You're invited ;)

Reply:
Pushed! :)
Thanks!

https://github.com/v1r0x/files_3d/pull/10/commits/2bd4cbd81be5b692cc27a484d03a0ae454a55292

Reply:
thanks for the fixes. But it still doesn't get loaded in files :(

Reply:
Make sure it's enabled :p
For me it's in the source code 
![Capture d’écran_2020-04-16_14-48-09](https://user-images.githubusercontent.com/14975046/79457921-4d5ce180-7ff1-11ea-9fc1-64b90972dcdb.png)

Reply:
:D I double checked that ;)
I also disabled caching in ff's dev console :confused:

Reply:
No errors in your nextcloud logs?

Reply:
Make sure you also have the viewer cloned & enabled  ^^
Are you developing on server@master ?

Reply:
I'm working on latest release (18.0.3). Viewer ist installed & enabled (v1.2.0) ;)
The only place where it's in the source code is here
```html
var _oc_appswebroots={..., "files_3d":"/nextcloud/apps/files_3d", ...}
```

Reply:
Weird! Try disabling and enabling it again?

Reply:
I replaced viewer with the stable18 branch from git and now it's no longer in the source code as well :confused:

Reply:
What are you using as a dev environment?
You cannot disable viewer, it's mandatory since 18 ;)

Reply:
I'm using latest nc stable release and cloned my app into /apps

Reply:
docker?

Reply:
No, old school ;) downloaded the zip and running on local apache (in /var/www/html)

Reply:
Yeah, it might not work, as the zip is a stable release and contains far more than just server and viewer :)

I would suggest you checkout the server repo and clone the viewer branch in apps :)

Reply:
It works :tada: I had to clone viewer as well, but now my js file is loaded :partying_face: 

thank you very much! Now I "simply" have to render the files ;)